lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -164,7 +164,7 @@ protected function doDelete($id)
     {
         $filename = $this->getFilename($id);
 
-        return @unlink($filename) || ! file_exists($filename);
```

Comment:
The idea is to always call `unlink()`, then check if the file exists, should `unlink()` fail.